packages/runtime/container-runtime/src/dataStoreContext.ts

```diff
@@ -800,6 +825,36 @@ export abstract class FluidDataStoreContext extends TypedEventEmitter<IFluidData
         }
     }
 
+    /**
+     * Summarizer client should not have local changes. These changes can become part of the summary and can break
+     * eventual consistency. For example, the next summary (say at ref seq# 100) may contain these changes whereas
+     * other clients that are up-to-date till seq# 100 may not have them yet.
+     */
+    protected identifyLocalChangeInSummarizer(eventName: string, type?: string) {
+        if (this.clientDetails.type === summarizerClientType) {
+            // If the count of telemetry logged has crossed the threshold, don't log any more.
+            if (this.localChangesTelemetryThreshold > 0) {
```

Comment:
nit: inconsistent in style - first IF vs. this IF (in terms of usage of early return statements)